src/components/apps/launch/ParamGroups.js

```diff
@@ -0,0 +1,375 @@
+        case constants.PARAM_TYPE.MULTILINE_TEXT:
+            return buildDebugId(baseParamId, ids.APP_LAUNCH_MULTI_LINE_TEXT);
+        case constants.PARAM_TYPE.REFERENCE_ANNOTATION:
+            return buildDebugId(
+                baseParamId,
+                ids.APP_LAUNCH_REFERENCE_ANNOTATION
+            );
+        case constants.PARAM_TYPE.REFERENCE_GENOME:
+            return buildDebugId(baseParamId, ids.APP_LAUNCH_REFERENCE_GENOME);
+        case constants.PARAM_TYPE.REFERENCE_SEQUENCE:
+            return buildDebugId(baseParamId, ids.APP_LAUNCH_REFERENCE_SEQUENCE);
+        case constants.PARAM_TYPE.TEXT:
+            return buildDebugId(baseParamId, ids.APP_LAUNCH_TEXT_INPUT);
+        case constants.PARAM_TYPE.TEXT_SELECTION:
+            return buildDebugId(baseParamId, ids.APP_LAUNCH_TEXT_SELECTION);
+        default:
+            return baseParamId;
+    }
+};
+
+/**
+ * Form fields and info display for an app parameter group.
+ */
+const ParamGroupForm = withI18N((props) => {
+    const classes = useStyles();
+
+    const {
+        baseId,
+        fieldName,
+        group,
+        referenceGenomes,
+        referenceGenomesLoading,
+        startingPath,
+    } = props;
+
+    return (
+        <ExpansionPanel id={baseId} defaultExpanded>
+            <ExpansionPanelSummary
+                expandIcon={
+                    <ExpandMore id={buildDebugId(baseId, ids.BUTTONS.EXPAND)} />
+                }
+            >
+                <Typography variant="subtitle1">{group.label}</Typography>
+            </ExpansionPanelSummary>
+            <ExpansionPanelDetails className={classes.expansionPanelDetails}>
+                {group.parameters?.map((param, paramIndex) => {
+                    if (!param.isVisible) {
+                        return null;
+                    }
+
+                    const name = `${fieldName}.parameters.${paramIndex}.value`;
+                    const paramFormId = buildParamId(
+                        baseId,
+                        paramIndex,
+                        param.type
+                    );
+
+                    let fieldProps = {
+                        id: paramFormId,
+                        name,
+                        label: param.label,
+                        required: param.required,
+                    };
+
+                    switch (param.type) {
+                        case constants.PARAM_TYPE.INFO:
+                            fieldProps = {
+                                id: paramFormId,
+                                name,
+                                component: Typography,
+                                variant: "body1",
+                                gutterBottom: true,
+                                dangerouslySetInnerHTML: {
+                                    __html: sanitizeHtml(param.label),
+                                },
+                            };
```

Comment:
Same Q. Why do we have to set inner html ?

Reply:
This text is coming from the app integrator, and in order to support a limited set of HTML (as we do in the current UI), we have to sanitize the string then set it as the inner HTML (again so the raw tags don’t display in the text).